packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -562,6 +563,22 @@ export class Viewer {
     return this.createScene();
   }
 
+  /**
+   * Waits for a certain duration for a frame with a given correlation ID to be present on a frame.
+   * Default timeout for waiting is 2 seconds, but can be configured.
+   */
+  @Method()
+  public async setWaitForFrameWithCorrelationId(
+    suppliedCorrelationId: string,
+    waitTimeoutMs = 2000
+  ): Promise<void> {
```

Comment:
Developers can either use `onFrameDrawn` or `onFrameReceived` event handlers and check if the frame has a correlation ID. I don't think we need to expose anything additional for this.